src/config.rs

```diff
@@ -362,6 +364,14 @@ impl Cfg {
         Toolchain::from(self, name)
     }
 
+    pub fn get_toolchains_from_regex(&self, regex: Regex) -> Result<Vec<Toolchain<'_>>> {
```

Comment:
Would it be possible to return a `Result<impl Iterator<Item=Toolchain<'_>>>` instead?  No need to `.collect()` in theory.

Reply:
Sure, then we could collect only when we need to.